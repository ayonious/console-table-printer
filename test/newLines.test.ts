import { Table } from '../index';

describe('Testing line breaks in table cells', () => {
  it('does not break the table', () => {
    // Create a table
    const p = new Table({
      columns: [
        { name: 'foo', alignment: 'right' },
        { name: 'bar', alignment: 'center', minLen: 40 },
        { name: 'baz', alignment: 'left', minLen: 40 },
      ],
    });

    p.addRows([
      {
        foo: 'Sed dictum',
        bar: 'Pellentesque\rhabitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Maecenas ut scelerisque felis. Etiam ut leo fringilla lacus posuere dignissim aliquet eu nulla.\n\nEtiam placerat ultricies sollicitudin. Maecenas ultrices feugiat tempor. Nam non nisi vel diam tristique commodo. Etiam sit amet nisi sagittis, fringilla tortor sed, efficitur nisl. Maecenas tincidunt lobortis dapibus.',
        baz: `* Cras ac ligula aliquet, ultrices erat quis, laoreet lorem.
* Mauris rutrum nisi et nisl sollicitudin semper.
* Nunc laoreet felis auctor diam aliquet semper.
* Integer aliquet urna imperdiet orci pulvinar, at vulputate ante consectetur.
* Pellentesque a lectus porttitor, condimentum lacus non, venenatis lacus.`,
      },
    ]);

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
