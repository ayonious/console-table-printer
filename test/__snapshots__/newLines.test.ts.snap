// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing line breaks in table cells does not break the table 1`] = `
"┌────────────┬──────────────────────────────────────────┬──────────────────────────────────────────┐
│[37m [0m[01m       foo[0m │[37m [0m[01m                  bar                   [0m │[37m [0m[01mbaz                                     [0m │
├────────────┼──────────────────────────────────────────┼──────────────────────────────────────────┤
│[37m [0m[37mSed dictum[0m │[37m [0m[37m              Pellentesque              [0m │[37m [0m[37m* Cras ac ligula aliquet, ultrices erat [0m │
│[37m [0m[37m          [0m │[37m [0m[37m  habitant morbi tristique senectus et  [0m │[37m [0m[37mquis, laoreet lorem.                    [0m │
│[37m [0m[37m          [0m │[37m [0m[37m   netus et malesuada fames ac turpis   [0m │[37m [0m[37m* Mauris rutrum nisi et nisl            [0m │
│[37m [0m[37m          [0m │[37m [0m[37megestas. Maecenas ut scelerisque felis. [0m │[37m [0m[37msollicitudin semper.                    [0m │
│[37m [0m[37m          [0m │[37m [0m[37m  Etiam ut leo fringilla lacus posuere  [0m │[37m [0m[37m* Nunc laoreet felis auctor diam aliquet[0m │
│[37m [0m[37m          [0m │[37m [0m[37m      dignissim aliquet eu nulla.       [0m │[37m [0m[37msemper.                                 [0m │
│[37m [0m[37m          [0m │[37m [0m[37m                                        [0m │[37m [0m[37m* Integer aliquet urna imperdiet orci   [0m │
│[37m [0m[37m          [0m │[37m [0m[37m Etiam placerat ultricies sollicitudin. [0m │[37m [0m[37mpulvinar, at vulputate ante consectetur.[0m │
│[37m [0m[37m          [0m │[37m [0m[37m Maecenas ultrices feugiat tempor. Nam  [0m │[37m [0m[37m* Pellentesque a lectus porttitor,      [0m │
│[37m [0m[37m          [0m │[37m [0m[37m  non nisi vel diam tristique commodo.  [0m │[37m [0m[37mcondimentum lacus non, venenatis lacus. [0m │
│[37m [0m[37m          [0m │[37m [0m[37mEtiam sit amet nisi sagittis, fringilla [0m │[37m [0m[37m                                        [0m │
│[37m [0m[37m          [0m │[37m [0m[37m  tortor sed, efficitur nisl. Maecenas  [0m │[37m [0m[37m                                        [0m │
│[37m [0m[37m          [0m │[37m [0m[37m      tincidunt lobortis dapibus.       [0m │[37m [0m[37m                                        [0m │
└────────────┴──────────────────────────────────────────┴──────────────────────────────────────────┘"
`;
